server/routes/functionalSkills/functionalSkillsController.ts

```diff
@@ -14,18 +18,35 @@ export default class FunctionalSkillsController {
       prisonNumber,
       req.user.username,
     )
-    const latestFunctionalSkillsFromCurious = mostRecentFunctionalSkills(functionalSkillsFromCurious)
-    const allFunctionalSkillsFromCurious = allFunctionalSkills(functionalSkillsFromCurious)
 
     const { problemRetrievingData } = functionalSkillsFromCurious
     const englishSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'ENGLISH')
     const mathsSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'MATHS')
     const digitalSkills = functionalSkillsByType(functionalSkillsFromCurious.assessments, 'DIGITAL_LITERACY')
 
+    // Handle the prison name lookup for each assessment
```

Comment:
This isn't working as you expect it is - you are creating and modifying a new object (`allSkillsByType`). Within that object you are mutating the assessment records via the `addPrisonNameToAssessment` function.
But .... you are not then using that modified `allSkillsByType` object in the view ..... but it is still working .. ? .. 🤔 

This is the crux of the eslint warning re: no-param-reassign ....

What you are actually doing is adding all the assessments from `englishSkills`, `mathsSkills` and `digitalSkills` to the new array `allSkillsByType`. And then the `addPrisonNameToAssessment` is iterating through those assessments (which are the same objects as the ones in the source arrays - they are not copies / new objects) and modifying them.

So, when you then add `englishSkills`, `mathsSkills` and `digitalSkills` to the view, they contain the modified assessments with the prison name.

Whilst it works, its not the way we want to do this.

I've just committed a change to this section of code that does it a better way. I've also updated the unit test to ensure that we are testing that the prison service is being called and that it's returned data is used in the resultant assessment objects 👍